File: modified_scripts/javascript/bubble_sort/deepseek-coder_1.3b.js

function bubble_sort(arr) {
    var len = arr.length;
    while(len > 0) {
        len = 0;
        var i=0;
        for(i; i < len-1 ; ++) {
            if(arr[i] < arr[i+1]) {
                var t =arr [i];
                arr[i] =arr [i+1];
                arr[i + 1] = t;
            }
        }
        --len;
    }
    return arr;
};

File: modified_scripts/rust/bubble_sort/deepseek-coder_1.3b.rs

pub fn bubble_sort<T: Ord>(arr: &mut [T]) {
    let len = arr.len();
     for i in (0..len).rev() { 
         for j in 1..=i {   
              if !arr[j - 1].gt(&arr[j]) { 
                  arr.swap(j - 1, j); 
              }  
         } 
     }   
}      